js/services/MeteorService.js

```diff
@@ -1,40 +1,54 @@
-const { previousMondayDate, previousFridayDate } = require('../utils/DateUtils')
 const requestUtils = require('../clients/MeteorClient')
+const MeteorPerDateDto = require('../dtos/MeteorPerDateDto')
 const meteorMapper = require('../mappers/MeteorMapper')
 
-const startDate = previousMondayDate
-const endDate = previousFridayDate
-
-const getMeteorsData = async () => {
-    const meteorsDataResponse = await requestUtils.getMeteorsWithinPeriod(startDate, endDate)
+const getMeteorsData = async (meteor) => {
+    const meteorDto = specifyDates(meteor)
+    const meteorsDataResponse = await requestUtils.getMeteorsWithinPeriod(meteorDto.startDate, meteorDto.endDate)
     const nearEarthObjects = meteorsDataResponse.data.near_earth_objects
 
-    return buildMeteorsDataResponse(nearEarthObjects)
+    return buildMeteorsDataResponse(nearEarthObjects, meteorDto)
 }
 
-const buildMeteorsDataResponse = (nearEarthObjects) => {
+const buildMeteorsDataResponse = (nearEarthObjects, updatedMeteor) => {
     if (nearEarthObjects === undefined) {
         return { data: {} }
     }
 
+    let wereDangerous = undefined
     const meteorsDataResponse = []
     Object.keys(nearEarthObjects).forEach((date) => {
-
+        const startCount = 0
         const meteors = []
-        nearEarthObjects[date].forEach((meteorStat) => {
-            meteors.push(meteorMapper.buildMeteorEntity(meteorStat))
-        })
+        nearEarthObjects[date]
+            .slice(startCount, updatedMeteor.count)
+            .forEach((meteorStat) => {
+                meteors.push(meteorMapper.buildMeteorEntity(meteorStat))
+                if (updatedMeteor.wereDangerous !== undefined && wereDangerous !== true) {
+                    wereDangerous = meteorStat.is_potentially_hazardous_asteroid
+                }
+            })
+        meteorsDataResponse.push(new MeteorPerDateDto(date, meteors))
+    })
 
-        const meteorsPerDate = {
-            data: {
-                date: date,
-                meteors: meteors
-            }
+    return {
+        data: {
+            mete: meteorsDataResponse,
+            were_dangerous: wereDangerous
         }
-        meteorsDataResponse.push(meteorsPerDate)
-    })
+    }
+}
 
-    return meteorsDataResponse
+const specifyDates = (meteorDto) => {
```

Comment:
As you use DTO, you can prepare data there - check if values are set and cast it to types you need
Another approach to use is Joi library - it can cast values according to schema while validation. I believe you can implement it in future PR's

Reply:
fixed. I will try to implement Joi library with the next dto from the next tasks